controller/Battery.php

```diff
@@ -85,6 +88,7 @@ public function create()
 
         $label = $this->getClassService()->createUniqueLabel($clazz);
         $item = $this->getClassService()->createInstance($clazz, $label);
+        $this->getEventManager()->trigger(new BatteryCreatedEvent($item, [$label]));
```

Comment:
consider to move event triggering to `taoBattery/model/service/rdf/RdfBatteryClassService.php` (like BatteryRemovedEvent, BatteryRemoveFailedEvent, BatteryBeforeRemoveEvent) to handle cases with direct `RdfBatteryClassService::createInstance()` call

Reply:
So I should move the whole method of creating an instance, in that case, and this refactoring was not part of this task.

Reply:
> So I should move the whole method of creating an instance

I mean overload `createInstance()` in `RdfBatteryClassService`:
```php
public function createInstance(core_kernel_classes_Class $clazz, $label = '')
    {
        $item = parent::createInstance($clazz, $label);
        $this->getEventManager()->trigger(new BatteryCreatedEvent($item, [$label]));
        return $item;
    }
```

The same approach I used for `deleteResource`. In your case event gets triggered only if instance created using controller method (this behavior differs with `BatteryRemovedEvent`).

Reply:
This RdfBatteryClassService is extending a deprecated class. And the whole idea seems not very pretty. I would leave it that way and consider refactoring all mechanism here instead modifying its elements. You are absolutely right though.